src/blocking/spi.rs

```diff
@@ -104,3 +104,28 @@ pub mod write_iter {
         }
     }
 }
+
+/// Operation for transactional SPI trait
+///
+/// This allows composition of SPI operations into a single bus transaction
+#[cfg(feature = "unproven")]
+#[derive(Debug, PartialEq)]
+pub enum Operation<'a, W: 'static> {
+    /// Write data from the provided buffer, discarding read data
+    Write(&'a [W]),
+    /// Write data from one buffer while reading into the second buffer
+    WriteRead(&'a [W], &'a mut [W]),
```

Comment:
WriteRead taking two arrays (as is consistent with the linux `spidev`, though one is also allowable) makes it difficult to map between `WriteRead` objects and the `Transfer` function, it's probably more useful to run a single array here instead...

Reply:
By `Transfer` you mean `exec` or something else?

Reply:
the [existing](https://github.com/rust-embedded/embedded-hal/blob/371251e03142f969d671b28f563efa695c5bc64a/src/blocking/spi.rs#L4) `spi::Transfer` trait. I have now swapped this to match (because otherwise polyfilling `Transactional` requires allocation).

Reply:
right, a side effect of this is that it becomes nearly impossible to wrap the linux spidev transactions :cry:, so, possibility of allocation it is i guess?

see [here](https://github.com/rust-embedded/linux-embedded-hal/pull/33/files#diff-b4aea3e418ccdb71239b96952d9cddb6R307):
``` rust
// Map types from generic to linux objects
let mut messages: Vec<_> = operations.as_mut().iter_mut().map(|a| {
  match a {
    SpiOperation::Write(w) => SpidevTransfer::write(w),
    SpiOperation::WriteRead(w, r) => SpidevTransfer::read_write(w, r) ,
  }
}).collect();
```

Reply:
It's possible to hack around this with:
``` rust
let w = unsafe {
  let p = r.as_ptr();
  std::slice::from_raw_parts(p, r.len())
};

SpidevTransfer::read_write(w, r)
```

however, it's not super clear from the [docs](https://www.kernel.org/doc/html/v4.14/driver-api/spi.html#c.spi_transfer) whether sharing tx_buffer and rx_buffer is okay

Reply:
Hmm... it does not look very nice. For the Linux kernel [implementation](https://github.com/torvalds/linux/blob/aae1464f46a2403565f75717438118691d31ccf1/drivers/spi/spidev.c#L198) I think it would work, as the buffers are copied in and out.